src/Sylius/Behat/Context/Setup/ProductContext.php

```diff
@@ -170,6 +188,88 @@ public function productVariantBelongsToTaxCategory(
     }
 
     /**
+     * @Given /^(this product) has ([^"]+) attribute "([^"]+)" with value "([^"]+)"$/
+     */
+    public function thisProductHasAttributeWithValue(ProductInterface $product, $productAttributeType, $productAttributeName, $value)
+    {
+        $this->createProductAttribute($productAttributeType,$productAttributeName);
+        $this->createProductAttributeValue($value);
+        $product->addAttribute($this->sharedStorage->get('product_attribute_value'));
```

Comment:
We can, but in three steps we need to create a `attributeValue`. The steps are very common, the only difference is that a specific value type should be passed to `setValue`. So there will be some repetitions in code. 
Maybe just omit adding attribute to shared storage and the `createProductAttributeValue` will be return an object? Or move whole logic from this method to steps mentioned above.